src/main/java/MyCallableForkJoinTask.java

```diff
@@ -0,0 +1,41 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.RecursiveTask;
+
+public class MyCallableForkJoinTask extends RecursiveTask<Integer> {
+    private static final int THRESHOLD = 20;
+    private List<Integer> list;
+
+    public MyCallableForkJoinTask(List<Integer> list) {
+        this.list = list;
+    }
+
+    @Override
+    protected Integer compute() {
+        if (list.size() > THRESHOLD) {
+            return ForkJoinTask.invokeAll(createSubtasks())
+                    .stream()
+                    .mapToInt(ForkJoinTask::join)
+                    .sum();
+        } else {
+            return processing(list);
+        }
```

Comment:
```suggestion
        if (list.size() > THRESHOLD) {
            return ForkJoinTask.invokeAll(createSubtasks())
                    .stream()
                    .mapToInt(ForkJoinTask::join)
                    .sum();
        }
        return processing(list);
```
Don’t complicate if-else construction.